File: examples/python/core/callbacks/message-broker-notification-handler.py
# Message processor side (separate Lambda or service)
import boto3
import json

lambda_client = boto3.client('lambda')

def process_payment_message(event: dict):
    """Process payment and notify callback."""
    callback_id = event["callback_id"]
    amount = event["amount"]
    customer_id = event["customer_id"]
    
    try:
        # Process payment with external system
        result = payment_processor.charge(customer_id, amount)
        
        # Notify success
        result_data = json.dumps({
            'status': 'completed',
            'transaction_id': result.transaction_id,
        }).encode('utf-8')
        
        lambda_client.send_durable_execution_callback_success(
            CallbackId=callback_id,
            Result=result_data
        )
    except PaymentError as e:
        # Notify failure
        lambda_client.send_durable_execution_callback_failure(
            CallbackId=callback_id,
            Error={
                'ErrorType': 'PaymentError',
                'ErrorMessage': f'{e.error_code}: {str(e)}'
            }
        )

File: examples/java/core/callbacks/message-broker-notification-handler.java
// Message processor side (separate Lambda or service)
import com.fasterxml.jackson.databind.ObjectMapper;
import software.amazon.awssdk.core.SdkBytes;
import software.amazon.awssdk.services.lambda.LambdaClient;
import software.amazon.awssdk.services.lambda.model.*;

LambdaClient lambdaClient = LambdaClient.create();
ObjectMapper mapper = new ObjectMapper();

public void processPaymentMessage(Map<String, Object> event) {
    String callbackId = (String) event.get("callback_id");
    Number amount = (Number) event.get("amount");
    String customerId = (String) event.get("customer_id");

    try {
        // Process payment with external system
        var result = paymentProcessor.charge(customerId, amount);

        // Notify success
        byte[] resultData = mapper.writeValueAsBytes(Map.of(
            "status", "completed",
            "transaction_id", result.getTransactionId()
        ));

        lambdaClient.sendDurableExecutionCallbackSuccess(
            SendDurableExecutionCallbackSuccessRequest.builder()
                .callbackId(callbackId)
                .result(SdkBytes.fromByteArray(resultData))
                .build());
    } catch (PaymentException e) {
        // Notify failure
        lambdaClient.sendDurableExecutionCallbackFailure(
            SendDurableExecutionCallbackFailureRequest.builder()
                .callbackId(callbackId)
                .error(CallbackError.builder()
                    .errorType("PaymentError")
                    .errorMessage(e.getErrorCode() + ": " + e.getMessage())
                    .build())
                .build());
    }
}
